Add office hours resources (#1698)
mozilla/extension-workshop: src/content/community.md

Stay up to date with the latest information about add-on technology on the [Add-ons blog](https://blog.mozilla.org/addons).

### Firefox for Android office hours <span id="office-hours"></span>

The Firefox Add-ons team holds one-on-one office hours sessions to help you make your extensions available on Android. This is in addition to the [Firefox for Android Add-ons](https://discourse.mozilla.org/c/add-ons/android/9393) forum, where you can seek advice and assistance anytime.

<details>
  <summary>Book a session</summary>

>   <iframe id="booking-frame" src="https://calendar.google.com/calendar/appointments/schedules/AcZssZ3yNnoZNBdJCTiJjqziXs9GoCYpfW9XZFkB0QWKNy9dfoqqPlCw7UsYh48Ff0i3lZcA5YzygwrU?gv=true" sandbox="allow-same-origin allow-scripts allow-forms allow-popups" style="height: 600px;"></iframe>

Reviewer: Just wondering - do we need to add `allow-top-navigation` or even `allow-top-navigation-by-user-activation` as well? Only if a navigation triggered from the calendar is an expected part of the functionality.

And maybe also allow-modals if you want to enable things like alert/confirm.

Author: The default sandbox value set by the Google Calendar scripts scripts is `"allow-same-origin allow-scripts allow-forms allow-popups allow-popups-to-escape-sandbox allow-storage-access-by-user-activation"`. I tried to trim this list to minimize the capabilities being granted to the sandbox. In retrospect, I _think_ those additional capabilities are reasonable and will therefore go back to the default sandbox settings.

I don't think we need to add navigation-related capabilities since Calendar itself didn't request them.

Stay up to date with the latest information about add-on technology on the [Add-ons blog](https://blog.mozilla.org/addons).

### Firefox for Android office hours <span id="office-hours"></span>

The Firefox Add-ons team holds one-on-one office hours sessions to help you make your extensions available on Android. This is in addition to the [Firefox for Android Add-ons](https://discourse.mozilla.org/c/add-ons/android/9393) forum, where you can seek advice and assistance anytime.

<details>
  <summary>Book a session</summary>

  <iframe id="booking-frame" src="https://calendar.google.com/calendar/appointments/schedules/AcZssZ3yNnoZNBdJCTiJjqziXs9GoCYpfW9XZFkB0QWKNy9dfoqqPlCw7UsYh48Ff0i3lZcA5YzygwrU?gv=true" sandbox="allow-same-origin allow-scripts allow-forms allow-popups allow-popups-to-escape-sandbox allow-storage-access-by-user-activation" style="height: 600px;"></iframe>
</details>

> <link href="https://calendar.google.com/calendar/scheduling-button-script.css" rel="stylesheet">
> <script src="https://calendar.google.com/calendar/scheduling-button-script.js" async></script>
> <script>
> (function() {
>   var target = document.currentScript;
>   window.addEventListener('load', function() {
>     calendar.schedulingButton.load({
>       url: 'https://calendar.google.com/calendar/appointments/schedules/AcZssZ3yNnoZNBdJCTiJjqziXs9GoCYpfW9XZFkB0QWKNy9dfoqqPlCw7UsYh48Ff0i3lZcA5YzygwrU?gv=true',
>       color: '#039BE5',
>       label: 'Book an appointment',
>       target,
>     });
>   });
> })();
> </script>
> 

Reviewer: ```suggestion
```

Rebase went wrong? Since the frame fulfills the role of the scheduler, let's remove the style and script.

Author: Oh my, yes. Something got off track. I backed up the branch before squashing, I'll get this fixed and re-pushed shortly.

Stay up to date with the latest information about add-on technology on the [Add-ons blog](https://blog.mozilla.org/addons).

> ### Office hours <span id="office-hours"></span>

Reviewer 2: As this is only for Firefox for Android
```suggestion
### Firefox for Android office hours" <span id="office-hours"></span>
```

> The Firefox Add-ons team holds one-on-one office hours sessions to help developers make their extensions available on Android. Developres can also use the [Firefox for Android Add-ons](https://discourse.mozilla.org/c/add-ons/android/9393) forum to discuss issues asynchronously.

Reviewer 3: Typo: Developres => Developers

Author: Fixed and force-pushed the update.

> The Firefox Add-ons team holds one-on-one office hours sessions to help developers make their extensions available on Android. Developers can also use the [Firefox for Android Add-ons](https://discourse.mozilla.org/c/add-ons/android/9393) forum to discuss issues asynchronously.

Reviewer 2: Make this more personal:
```suggestion
The Firefox Add-ons team holds one-on-one office hours sessions to help you make your extensions available on Android. This is in addition to the [Firefox for Android Add-ons](https://discourse.mozilla.org/c/add-ons/android/9393) forum, where you can seek advice and assistance anytime.
```

<!-- Google Calendar Appointment Scheduling begin -->
<link href="https://calendar.google.com/calendar/scheduling-button-script.css" rel="stylesheet">

> <script src="https://calendar.google.com/calendar/scheduling-button-script.js" async></script>

Reviewer 4: I don't think this will work once deployed because `https://calendar.google.com/calendar/` isn't specified in the CSP.

Author: Good call. I'll revise the PR with this and [Rob's comment](https://github.com/mozilla/extension-workshop/pull/1698#issuecomment-1726184721) in mind.

<script>
(function() {
  var target = document.currentScript;
  window.addEventListener('load', function() {
    calendar.schedulingButton.load({
      url: 'https://calendar.google.com/calendar/appointments/schedules/AcZssZ3yNnoZNBdJCTiJjqziXs9GoCYpfW9XZFkB0QWKNy9dfoqqPlCw7UsYh48Ff0i3lZcA5YzygwrU?gv=true',
      color: '#039BE5',

>       label: 'Book an appointment',

Reviewer 2: To align with earlier wording
```suggestion
      label: 'Book a session',
```

Author: This widget has been dropped in response to @Rob--W's feedback. The language used to open the new UI element is "Book a session".

      target,
    });
  });
})();
</script>
<!-- end Google Calendar Appointment Scheduling -->
<style>
/*
 * Google's scripts inject a button styled with "Roboto" but doesn't load that
 * font. Unfortunately, that means the font-family falls back to Times in
 * Firefox and Chrome. Since we can't set an inline style on this element, the
 * next best thing we can do is force the style on buttons in this this section.
 */
#connect-with-the-community button {
  font-family: inherit !important;
}

> </style>

Reviewer 4: It would be good to move this to a scss file:

```diff
diff --git a/src/assets/css/_hacks.scss b/src/assets/css/_hacks.scss
new file mode 100644
index 0000000..423aca7
--- /dev/null
+++ b/src/assets/css/_hacks.scss
@@ -0,0 +1,9 @@
+/**
+ * Google's scripts inject a button styled with "Roboto" but doesn't load that
+ * font. Unfortunately, that means the font-family falls back to Times in
+ * Firefox and Chrome. Since we can't set an inline style on this element, the
+ * next best thing we can do is force the style on buttons in this this section.
+ */
+#connect-with-the-community button {
+  font-family: inherit;
+}
diff --git a/src/assets/css/styles.scss b/src/assets/css/styles.scss
index b3735ec..390d6bf 100644
--- a/src/assets/css/styles.scss
+++ b/src/assets/css/styles.scss
@@ -20,6 +20,7 @@ CSS file.
 @import 'contentguidelines';
 @import 'fullwidth';
 @import 'sidebar';
+@import 'hacks';
 
 // Select the components you want to use
 @include foundation-global-styles;
diff --git a/src/content/community.md b/src/content/community.md
index 5839c8a..3c99eae 100644
--- a/src/content/community.md
+++ b/src/content/community.md
@@ -66,7 +66,7 @@ The Firefox Add-ons team holds one-on-one office hours sessions to help develope
   window.addEventListener('load', function() {
     calendar.schedulingButton.load({
       url: 'https://calendar.google.com/calendar/appointments/schedules/AcZssZ3yNnoZNBdJCTiJjqziXs9GoCYpfW9XZFkB0QWKNy9dfoqqPlCw7UsYh48Ff0i3lZcA5YzygwrU?gv=true',
-      color: '#039BE5',
+      color: '#015fdf',
       label: 'Book an appointment',
       target,
     });
@@ -74,17 +74,6 @@ The Firefox Add-ons team holds one-on-one office hours sessions to help develope
 })();
 </script>
 <!-- end Google Calendar Appointment Scheduling -->
-<style>
-/*
- * Google's scripts inject a button styled with "Roboto" but doesn't load that
- * font. Unfortunately, that means the font-family falls back to Times in
- * Firefox and Chrome. Since we can't set an inline style on this element, the
- * next best thing we can do is force the style on buttons in this this section.
- */
-#connect-with-the-community button {
-  font-family: inherit !important;
-}
-</style>
 
 ### Questions about add-on development
 
```

Also the color of the button should probably be like the other one at the top.

Author: These suggestions have been integrated.
